discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -0,0 +1,305 @@
+                    logger.info(f"index_solana_plays.py | sliced tx_sigs to {len(transaction_signatures)} entries")
+
+                # Reset batch state
+                transaction_signature_batch = []
+        logger.info(
+            f"index_solana_plays.py | intersection_found={intersection_found}, last_tx_signature={last_tx_signature}"
+        )
+
+    logger.info(f"index_solana_plays.py | {transaction_signatures}, {len(transaction_signatures)} entries")
+
+    transaction_signatures.reverse()
+
+    logger.info(f"index_solana_plays.py | {transaction_signatures}")
+
+    num_txs_processed = 0
+
+    for tx_sig_batch in transaction_signatures:
+        logger.error(f"index_solana_plays.py | processing {tx_sig_batch}")
+        batch_start_time = time.time()
+        # Process each batch in parallel
+        with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+            with db.scoped_session() as session:
+                parse_sol_tx_futures = {
+                    executor.submit(parse_sol_play_transaction, session,
+                                    solana_client, tx_sig): tx_sig
+                    for tx_sig in tx_sig_batch
+                }
+                for future in concurrent.futures.as_completed(
+                        parse_sol_tx_futures):
+                    try:
+                        # No return value expected here so we just ensure all futures are resolved
+                        future.result()
+                        num_txs_processed += 1
+                    except Exception as exc:
+                        logger.error(f"index_solana_plays.py | {exc}")
+
+        batch_end_time = time.time()
+        batch_duration = batch_end_time - batch_start_time
+        logger.info(
+            f"index_solana_plays.py | processed batch {len(tx_sig_batch)} txs in {batch_duration}s"
+        )
+
+    # Update plays iff some batch is found
+    if num_txs_processed > 0:
+        logger.info(
+            f"index_solana_plays.py | processed {num_txs_processed} txs"
+        )
+        with db.scoped_session() as session:
+            logger.info(
+                f"index_solana_plays.py | Refreshing aggregate_plays materialized view"
+            )
+            session.execute("REFRESH MATERIALIZED VIEW CONCURRENTLY aggregate_plays")
+            logger.info(
+                f"index_solana_plays.py | Finished refreshing aggregate_plays materialized view"
+            )
+
+######## CELERY TASKS ########
+@celery.task(name="index_solana_plays", bind=True)
+def index_solana_plays(self):
+    # Cache custom task class properties
+    # Details regarding custom task context can be found in wiki
+    # Custom Task definition can be found in src/__init__.py
+    redis = index_solana_plays.redis
+    solana_client = index_solana_plays.solana_client
+    # Define lock acquired boolean
+    have_lock = False
+    # Define redis lock object
+    update_lock = redis.lock("solana_plays_lock", timeout=7200)
```

Comment:
context for the 7200?

Reply:
7200 in this context represents the [maximum lifetime of the lock](https://github.com/andymccurdy/redis-py/blob/1a41cfd95a53a95b078084d8627be6b6fba3bb71/redis/lock.py#L79), which is 2 hours and also used in various places throughout other indexing tasks - but I am figuring out whether this is appropriate here as well

Reply:
updated to match core `index.py` w/`blocking_timeout=25`